sale_order_currency_rate/models/res_config_settings.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2023 Jarsa
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    sale_show_currency_rate = fields.Selection(
```

Comment:
```suggestion
    sale_show_currency_rate = fields.Selection(related="company_id.sale_show_currency_rate")
```

You don't need to add the selections and the string, it will inherit from the relation